src/interface/emacs/khoj.el

```diff
@@ -416,10 +416,47 @@ Auto invokes setup steps on calling main entrypoint."
          (files-to-index (or file-paths
                              (append (mapcan (lambda (dir) (directory-files-recursively dir "\\.\\(org\\|md\\|markdown\\|pdf\\|txt\\|rst\\|xml\\|htm\\|html\\)$")) content-directories) content-files)))
          (type-query (if (or (equal content-type "all") (not content-type)) "" (format "t=%s" content-type)))
+         (delete-files (khoj--get-delete-file-list khoj--indexed-files files-to-index))
          (inhibit-message t)
-         (message-log-max nil))
-    (let ((url-request-method "POST")
-          (url-request-data (khoj--render-files-as-request-body files-to-index khoj--indexed-files boundary))
+         (message-log-max nil)
+         (batch-size 30))
+	(dolist (files (khoj--split-file-list files-to-index batch-size))
+      (khoj--send-index-update-request (khoj--render-update-files-as-request-body files boundary) boundary content-type type-query force))
+    (when delete-files
+        (khoj--send-index-update-request (khoj--render-delete-files-as-request-body delete-files boundary) boundary content-type type-query force))
+    (setq khoj--indexed-files files-to-index)))
+
+(defun khoj--get-delete-file-list (indexed-files upload-files)
+  "Get delete file list. when `INDEXED-FILES' no longer in `UPLOAD-FILES'.
+delete them. return delete-file-list."
+  (let (delete-files '())
+    (dolist (indexed-file indexed-files)
+      (when (not (member indexed-file upload-files))
+        (push indexed-file delete-files)))
+    delete-files))
+
+(defun khoj--split-file-list (file-list size)
```

Comment:
Should we simplify the batching of `file-list` into subgroups of `size` by using the `dash.el` [-partition-all](https://github.com/magnars/dash.el?tab=readme-ov-file#-partition-all-n-list) function? It'll be simplify/replace this function with something like `(-partition-all size file-list)`

Reply:
Okay, I'll try this. I haven't used this function before